src/batch/src/executor/monitor/stats.rs

```diff
@@ -46,12 +44,189 @@ impl BatchMetrics {
+                                unimplemented!("Never reach here");
+                            }
+                        }
+                    }
+                }
+
+                // read from delete queue and push into delete_cache.
+                loop {
+                    match delete_queue_receiver.try_recv() {
+                        Ok(delete_record) => {
+                            delete_cache.push(delete_record);
+                        }
+                        Err(tokio::sync::mpsc::error::TryRecvError::Empty) => {
+                            break;
+                        }
+                        Err(_) => {
+                            // Error handle need modify later.
+                            error!("delete_queue_receiver is closed");
+                            connect = false;
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
         Self {
-            row_seq_scan_next_duration,
+            registry,
+            delete_queue_sender,
             exchange_recv_row_number,
         }
     }
 
+    pub fn create_task_metrics(&self, query_id: String) -> BatchTaskMetrics {
+        BatchTaskMetrics {
+            query_id,
+            record: RwLock::new(HashMap::new()),
+            exchange_recv_row_number: self.exchange_recv_row_number.clone(),
+            delete_queue_sender: self.delete_queue_sender.clone(),
+        }
+    }
+
+    /// Create a new `BatchTaskMetricsManager` instance used in tests or other places.
+    pub fn unused() -> Self {
+        let (delete_queue_sender, _) = tokio::sync::mpsc::channel::<DeleteRecord>(1);
+        Self {
+            registry: Registry::new(),
+            delete_queue_sender,
+            exchange_recv_row_number: register_int_counter_vec_with_registry!(
+                "batch_exchange_recv_row_number",
+                "Total number of row that have been received from upstream source",
+                &[
+                    "query_id",
+                    "source_stage_id",
+                    "target_stage_id",
+                    "source_task_id",
+                    "target_task_id"
+                ],
+                Registry::new()
+            )
+            .unwrap(),
+        }
+    }
+}
+
+pub struct BatchTaskMetrics {
+    pub query_id: String,
+
+    /// Used to delete the label haved been recorded.
+    /// record definition:
+    /// HashMap { Key: metrics_name ,
+    ///           Value: the list of labels [ source_stage_id,
+    /// target_stage_id,source_task_id,target_task_id]         }
+    pub record: RwLock<HashMap<String, Vec<[u32; 4]>>>,
+    pub exchange_recv_row_number: GenericCounterVec<AtomicU64>,
+    /// Used to send DeleteRecor to delete_queue.
+    delete_queue_sender: Sender<DeleteRecord>,
+}
+
+#[derive(Debug)]
+pub struct DeleteRecord {
+    query_id: String,
+    records: HashMap<String, Vec<[u32; 4]>>,
```

Comment:
Maybe make the Vector into a struct is better...

Reply:
Do you mean HashMap<String, Vec<Vec<u32>>>?

Reply:
```rust
// Minimal unit for each delete
LabelDeleteUnit {
target_stage_id,
source_stage_id,
target_task_id,
source_task_id
}
```